apps/postybirb-ui/src/app/app-i18n-provider.tsx

```diff
@@ -22,29 +30,61 @@ export function AppI18nProvider(props: AppI18nProviderProps) {
       // eslint-disable-next-line no-param-reassign
       locale = locale ?? 'en';
       const { messages } = await import(`../lang/${locale}.po`);
+
+      const uppyLocale = uppyLocales[locale];
+      try {
+        i18n.uppy = (
+          await import(`../../public/uppy-i18n/${uppyLocale}.js`)
+        ).default;
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(
+          // eslint-disable-next-line lingui/no-unlocalized-strings
+          `Failed to load uppy locale for ${locale}, mapped: ${uppyLocale}, error:`,
+          error
+        );
+      }
+
       i18n.loadAndActivate({ locale, messages });
       if (!loaded) setLoaded(true);
     },
     [loaded]
   );
 
   useEffect(() => {
-    SettingsStore.updates.subscribe((e) => {
+    const { unsubscribe } = SettingsStore.updates.subscribe((e) => {
       setLocale(e[0].settings.language);
     });
+
+    return unsubscribe;
   }, [setLocale]);
 
+  const [tooLongLoading, setTooLongLoading] = useState(false);
+  useEffect(() => {
+    setTimeout(() => {
+      if (!loaded) {
+        setTooLongLoading(true);
+      }
+    }, 5000);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   if (loaded) {
     return <LinguiI18nProvider i18n={i18n} {...props} />;
   }
 
   return (
-    <EuiFlexGroup alignItems="center" justifyContent="center">
-      <EuiLoadingSpinner size="xl" />
-      {
+    <EuiFlexGroup justifyContent="center" style={{ alignContent: 'center' }}>
+      <EuiFlexGroup justifyContent="center" alignItems="center">
+        <EuiLoadingSpinner size="xl" />
+        {
+          // eslint-disable-next-line lingui/no-unlocalized-strings
+        }
+        Loading translations...
+      </EuiFlexGroup>
+      {tooLongLoading &&
         // eslint-disable-next-line lingui/no-unlocalized-strings
-      }
-      Loading translations...
+        'Loading takes too much time, please check the console for the errors.'}
```

Comment:
This is for those edge cases when user could report us something like "Postybirb loading stuck at the loading translations"

So instead postybirb will show them what to do first and we will not need to say them to open the console